app/controllers/posts_controller.rb

```diff
@@ -25,6 +25,19 @@ def create
     end
   end
 
+  def destroy
+    @author = User.find(params[:user_id])
+    @post = @author.posts.find(params[:id])
+    @post.destroy
+    if @post.destroy
+      redirect_to user_path(params[:user_id])
+    else
+      redirect_to user_posts_path(params[:user_id], params[:id])
+    end
+  end
+
+  private
+
   def post_params
     params.require(:post).permit(:title, :text)
   end
```

Comment:
- [ ] Based on the requirements of this project, app should use CanCanCan for Post authorization in the Post Controller that's

```ruby
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

Let's kindly fix this issue.